CommunicationLayer/MessagingFramework/MotorDetailsMessage.cpp

```diff
@@ -0,0 +1,153 @@
+#include "MotorDetailsMessage.h"
+#include "MessageDecodingHelpers.h"
+#include "MessageDefines.h"
+
+#include <QDebug>
+
+using namespace MessageDecodingHelpers;
+
+namespace
+{
+    const int PHASE_CCURRENT = 1;
+    const int PHASE_BCURRENT = 5;
+    const int MOTOR_VOLTAGE_REAL = 9;
+    const int MOTOR_VOLTAGE_IMAGINARY = 13;
+    const int MOTOR_CURRENT_REAL = 17;
+    const int MOTOR_CURRENT_IMAGINARY = 21;
+    const int BACK_EMF_REAL = 25;
+    const int BACK_EMF_IMAGINARY = 29;
+    const int VOLTAGE_RAIL_SUPPPLY_15_V = 33;
+    const int VOLTAGE_RAIL_SUPPLY_33_V = 37;
+    const int VOLTAGE_RAIL_SUPPLY_19_V = 41;
+    const int HEAT_SINK_TEMPERATURE = 45;
+    const int MOTOR_TEMPTERATURE = 49;
+    const int DSP_BOARD_TEMPERATURE = 53;
+    const int DC_BUS_AMP_HOURS = 57;
+    const int ODOMETER = 61;
+    const int SLIP_SPEED = 65;
+}
+
+MotorDetailsMessage::MotorDetailsMessage(const QByteArray& messageData, unsigned char motorNumber)
+: messageData_(messageData)
+, motorNumber_(motorNumber)
```

Comment:
I didn't think that implementing this identical file for both motors made any sense, is this an appropriate way to solve this? Is there something other than an unsigned char I should be using for the motor number? Also does this need to be documented somewhere on the spreadsheet if it isn't external to any of the other programs?